cogeo_mosaic/backend/base.py

```diff
@@ -0,0 +1,70 @@
+"""cogeo_mosaic.backend.base: base Backend class."""
+
+import abc
+from contextlib import AbstractContextManager
+from typing import Dict, Optional
+
+from cogeo_mosaic import version as mosaic_version
+from cogeo_mosaic.backend.utils import get_hash
+from cogeo_mosaic.utils import create_mosaic
+
+
+class BaseBackend(AbstractContextManager):
+    mosaic_def: Optional[Dict]
+
+    @abc.abstractmethod
+    def __init__(self, *args, **kwargs):
+        """Connect to backend"""
+
+    def __enter__(self):
```

Comment:
Well my thought was that since `__enter__` always calls `__init__`, it's easiest to just have the default base `__enter__` and `__exit__` be non-abstract methods, so that the child classes don't actually need to overwrite `__enter__`, just `__init__`.

Reply:
> so that the child classes don't actually need to overwrite __enter__

If this is the case we should remove `__enter__` and `__exit__` from the base class because they are already defined in `abc.AbstractContextManager`.

Reply:
That's what I thought but then when I actually ran the code it gave me errors unless I specifically added `__enter__` and `__exit__` as methods.

Reply:
I think it's probably that `abc.AbstractContextManager` defines `__enter__` and `__exit__` as _abstract_ methods. Since we really don't have any setup/teardown, I think defining `__enter__` and `__exit__` as non-abstract methods in the base class are fine, and then we really don't need to subclass from `abc.AbstractContextManager` anyways